pkgs/development/perl-modules/generic/builder.sh

```diff
@@ -22,7 +22,12 @@ preConfigure() {
         fi
     done
 
-    perl Makefile.PL PREFIX=$out INSTALLDIRS=site $makeMakerFlags PERL=$(type -P perl) FULLPERL=\"$fullperl/bin/perl\"
+    # https://metacpan.org/pod/ExtUtils::MakeMaker#Static-Linking-of-a-new-Perl-Binary
+    if [ "$LINKTYPE" = "static" ]; then
+        perl Makefile.PL PREFIX=$out INSTALLDIRS=site $makeMakerFlags PERL=$(type -P perl) FULLPERL=\"$fullperl/bin/perl\" LINKTYPE=static
+    else
+        perl Makefile.PL PREFIX=$out INSTALLDIRS=site $makeMakerFlags PERL=$(type -P perl) FULLPERL=\"$fullperl/bin/perl\"
+    fi
```

Comment:
As @alyssais notes, we could also simplify this to

```suggestion
    LINKTYPE="${LINKTYPE:-dynamic}"
    perl Makefile.PL PREFIX=$out INSTALLDIRS=site $makeMakerFlags PERL=$(type -P perl) FULLPERL=\"$fullperl/bin/perl\" LINKTYPE=$LINKTYPE
```

since there's only[ two valid options](https://perldoc.perl.org/ExtUtils::MakeMaker#LINKTYPE) for `$LINKTYPE`.